components/Events/Filters.js

```diff
@@ -0,0 +1,204 @@
+          onKeyPress={(e) => e.key === 'Enter' && handleSearch()}
+        />
+        <button onClick={handleSearch} className={styles.searchbutton}>
+          search
+        </button>
+        <button
+          onClick={() => setIsFilterOpen(!isFilterOpen)}
+          className={styles.searchbutton}
+        >
+          filter
+          <FontAwesomeIcon
+            className={styles.arrow}
+            icon={isFilterOpen ? faChevronUp : faChevronDown}
+          />
+        </button>
+        {isFilterOpen && (
+          <div className={styles.filteroptions}>
+            <div>
+              <h3 className={styles.header}>Committees </h3>
+              {Object.keys(chosenComms).map((comm) => (
+                <div key={comm}>
+                  <input
+                    type="checkbox"
+                    id={comm}
+                    checked={chosenComms[comm]}
+                    onChange={() => handleCommChange(comm)}
+                  />{' '}
+                  {comm}
+                </div>
+              ))}
+            </div>
+            <div>
+              <h3 className={styles.header}>Days</h3>
+              {Object.keys(chosenDays).map((day) => (
+                <div key={day}>
+                  <input
+                    type="checkbox"
+                    id={day}
+                    checked={chosenDays[day]}
+                    onChange={() => handleWeekdayChange(day)}
+                  />{' '}
+                  {day}
+                </div>
+              ))}
+            </div>
+            <div>
+              <h3 className={styles.header}>Time</h3>
+              <div className={styles.timecontainer}>
+                <div>
+                  <div>From </div>
+                  <div>To </div>
+                </div>
+                <div>
+                  {[[fromTime, setFromTime],[toTime, setToTime]].map((state,index) => (
+                    <div key={index}>
+                      <input
+                        type="time"
+                        className={styles.timeinput}
+                        value={state[0]}
+                        onChange={(e) => state[1](e.target.value)}
+                      />
+                      <br />
+                    </div>
+                  ))}
+                </div>
+              </div>
+            </div>
+            <div>
+              <h3 className={styles.header}>Location</h3>
+              {Object.keys(chosenLoc).map((loc) => {
+                let title = loc;
+                if (loc === 'In_person') title = 'In-Person';
+                return (
+                  <div key={loc}>
+                    <input
+                      type="checkbox"
+                      id="online"
+                      checked={chosenLoc[loc]}
+                      onChange={() => handleLocationChange(loc)}
+                    />{title}
+                  </div>
+                );
+              })}
```

Comment:
This can be simplified if we change to the `In-Person` key + can we make the `id` equal to `loc` or something?